src/psyclone/psyir/backend/sir.py

```diff
@@ -0,0 +1,408 @@
+        PSyIR tree.
+
+        :param node: a Reference PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.Reference`
+
+        :returns: the SIR code as a string.
+        :rtype: str
+
+        :raises VisitorError: if this node has children.
+
+        '''
+        if node.children:
+            raise VisitorError(
+                "Method reference_node in class SIRWriter: SIR Reference "
+                "node is not expected to have any children.")
+        return "{0}makeVarAccessExpr(\"{1}\")".format(self._nindent, node.name)
+
+    def array_node(self, node):
+        '''This method is called when an Array instance is found in the PSyIR
+        tree.
+
+        :param node: an Array PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.Array`
+
+        :returns: the SIR code as a string.
+        :rtype: str
+
+        '''
+        stencil = gen_stencil(node)
+        result = ("{0}makeFieldAccessExpr(\"{1}\",{2})"
+                  "".format(self._nindent, node.name, stencil))
+        # _field_names is a set so duplicates will be ignored.
+        self._field_names.add(node.name)
+        return result
+
+    def literal_node(self, node):
+        '''This method is called when a Literal instance is found in the PSyIR
+        tree.
+
+        :param node: a Literal PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.Literal`
+
+        :returns: the SIR code as a string.
+        :rtype: str
+
+        '''
+        result = node.value
+        # There is an assumption here that the literal is a float (see
+        # #468).
+        return ("{0}makeLiteralAccessExpr(\"{1}\", BuiltinType.Float)"
+                "".format(self._nindent, result))
+
+    def unaryoperation_node(self, node):
+        '''This method is called when a UnaryOperation instance is found in
+        the PSyIR tree.
+
+        :param node: a UnaryOperation PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.UnaryOperation`
+
+        :returns: the SIR code as a string.
+        :rtype: str
+
+        :raises VisitorError: if there is no mapping from the PSyIR \
+        operator to SIR, or if the child of the PSyIR operator is not \
+        a literal (as only -<literal> is currently supported).
+
+        '''
+        # Currently only '-' is supported in the SIR mapping.
+        unary_operators = {
+            UnaryOperation.Operator.MINUS: '-'}
+        try:
+            oper = unary_operators[node.operator]
+        except KeyError:
+            raise VisitorError(
+                "Method unaryoperation_node in class SIRWriter, unsupported "
+                "operator '{0}' found.".format(str(node.operator)))
+        # Currently only '-<literal>' is supported in the SIR mapping.
+        if not (len(node.children) == 1 and
+                isinstance(node.children[0], Literal)):
+            raise VisitorError("Child of unary operator should be a literal.")
```

Comment:
Should this say that unary operations are only supported on literals rather than "should be"?

Reply:
Changed.